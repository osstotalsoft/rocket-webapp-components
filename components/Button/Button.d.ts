import { ButtonProps as BaseProps } from '@material-ui/core/Button';
import React from 'react';

export interface ButtonProps extends Omit<BaseProps, 'color' | 'size'> {
  /**
  * The color of the button.
  */
  color?: "primary" |
  "info" |
  "theme" |
  "themeNoBackground" |
  "themeWithBackground" |
  "success" |
  "warning" |
  "danger" |
  "rose" |
  "white" |
  "simple" |
  "defaultNoBackground" |
  "primaryNoBackground" |
  "infoNoBackground" |
  "successNoBackground" |
  "warningNoBackground" |
  "dangerNoBackground" |
  "roseNoBackground" |
  "white" |
  "transparent"
  /**
   * If true, rounded corners are enabled.
   */
  round?: boolean
  /**
  * 	The content of the component.
  */
  children?: React.ReactNode
  /**
  * 	If true, the button will take up the full width of its container.
  */
  fullWidth?: boolean
  /**
  * If true, the button will be disabled.
  */
  disabled?: boolean
  /**
  * Override or extend the styles applied to the component 
  * @deprecated Use `className` instead.
  */
  customClass?: string
  /**
  * A custom class provided.
  */
  className?: string
  /**
  * If true,  the button's min width will be set to 160px.
  */
  wd?: boolean
  /**
  * If true, the button will be smaller.
  */
  justIcon?: boolean
  /**
  * If true, the button will float to the right.
  */
  right?: boolean
  /**
  * The size of the button.
  */
  size?: "sm" | "lg" | "xs"
  /**
  * The tooltip of the button.
  */
  tooltip?: string
  /**
   * Element placed before the children.
   */
  startIcon?: React.ReactNode
  /**
   * Element placed after the children.
   */
  endIcon?: React.ReactNode
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/button
 *
 *   Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/Button/Button.js
 * 
 * Material-UI components used:
 * 
 * - Button: https://material-ui.com/api/button/
 * - Tooltip: https://material-ui.com/api/tooltip/
 * 
 */
export default function Button(props: ButtonProps): JSX.Element;