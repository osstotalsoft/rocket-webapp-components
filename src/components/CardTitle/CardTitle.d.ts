import { BoxProps } from "@material-ui/core";

export interface CardTitleProps extends BoxProps {
    /**
    * The title of card.
    */
    title?: string;
    /**
     * The actions(made of array of buttons) of card.
     */
    actions: React.ReactNode[]
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/card-title
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/CardTitle/CardTitle.js
 */
export default function CardTitle(props: CardTitleProps): JSX.Element;