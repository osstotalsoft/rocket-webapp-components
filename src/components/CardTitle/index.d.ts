export { default } from './CardTitle';
export * from './CardTitle';