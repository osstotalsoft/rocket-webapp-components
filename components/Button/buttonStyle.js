import styles from '../../assets/jss/styles';	

const buttonStyle = theme => {
  const {
    grayColor,
    whiteColor,
    hexToRgb,
    defaultFont,
    buttonColors
  } = styles(theme);

  return {
    button: {
      ...defaultFont,
      fontWeight: "bold",
      minHeight: "auto",
      minWidth: "auto",
      backgroundColor: grayColor,
      color: whiteColor,
      boxShadow:
        "0 2px 2px 0 rgba(" +
        hexToRgb(grayColor) +
        ", 0.14), 0 3px 1px -2px rgba(" +
        hexToRgb(grayColor) +
        ", 0.2), 0 1px 5px 0 rgba(" +
        hexToRgb(grayColor) +
        ", 0.12)",
      border: "none",
      borderRadius: "3px",
      position: "relative",
      padding: "12px 30px",
      margin: ".3125rem 1px",
      textTransform: "uppercase",
      letterSpacing: "0",
      willChange: "box-shadow, transform",
      transition: "box-shadow 0.2s cubic-bezier(0.4, 0, 1, 1), background-color 0.2s cubic-bezier(0.4, 0, 0.2, 1)",
      textAlign: "center",
      whiteSpace: "nowrap",
      verticalAlign: "middle",
      touchAction: "manipulation",
      cursor: "pointer",
      "&:hover,&:focus": {
        color: whiteColor,
        backgroundColor: grayColor,
        boxShadow:
          "0 14px 26px -12px rgba(" +
          hexToRgb(grayColor) +
          ", 0.42), 0 4px 23px 0px rgba(" +
          hexToRgb(grayColor) +
          ", 0.2)"
      },
      "& .fab,& .fas,& .far,& .fal,& .material-icons": {
        position: "relative",
        display: "inline-block",
        top: "0",
        marginTop: "-1em",
        marginBottom: "-1em",
        fontSize: "1.1rem",
        marginRight: "4px",
        verticalAlign: "middle"
      },
      "& svg": {
        position: "relative",
        display: "inline-block",
        top: "0",
        width: "25px",
        height: "25px",
        marginRight: "4px",
        verticalAlign: "middle"
      },
      "&$justIcon": {
        "& .fab,& .fas,& .far,& .fal,& .material-icons": {
          marginTop: "0px",
          position: "absolute",
          width: "100%",
          transform: "none",
          left: "0px",
          top: "0px",
          height: "100%",
          lineHeight: "41px",
          fontSize: "20px"
        }
      }
    },
    right: {
      float: "right"
    },
    fullWidth: {
      width: "100%"
    },
    primary: buttonColors.primary,
    info: buttonColors.info,
    success: buttonColors.success,
    warning: buttonColors.warning,
    danger: buttonColors.danger,
    rose: buttonColors.rose,
    white: buttonColors.white,
    simple: buttonColors.simple,
    defaultNoBackground: buttonColors.defaultNoBackground,
    primaryNoBackground: buttonColors.primaryNoBackground,
    infoNoBackground: buttonColors.infoNoBackground,
    successNoBackground: buttonColors.successNoBackground,
    warningNoBackground: buttonColors.warningNoBackground,
    dangerNoBackground: buttonColors.dangerNoBackground,
    roseNoBackground: buttonColors.roseNoBackground,
    theme: buttonColors.theme,
    themeNoBackground: buttonColors.themeNoBackground,
    themeWithBackground: buttonColors.themeWithBackground,
    disabled: {
      ...buttonColors.disabled,
      "&$justIcon": {
        pointerEvents: "none",
        color: whiteColor
      }
    },
    transparent: buttonColors.transparent,
    lg: {
      ...defaultFont,
      "&$justIcon": {
        "& .fab,& .fas,& .far,& .fal,& svg,& .material-icons": {
          marginTop: "-4px"
        }
      },
      padding: "1.125rem 2.25rem",
      fontSize: "14px",
      lineHeight: "1.333333",
      borderRadius: "0.2rem"
    },
    sm: {
      ...defaultFont,
      "&$justIcon": {
        "& .fab,& .fas,& .far,& .fal,& svg,& .material-icons": {
          marginTop: "1px"
        }
      },
      padding: "0.40625rem 1.25rem",
      fontSize: "12px",
      lineHeight: "1.5",
      borderRadius: "0.2rem"
    },
    xs: {
      ...defaultFont,
      padding: "4px 4px",
      fontSize: "10px"
    },
    round: {
      borderRadius: "30px"
    },
    block: {
      width: "100% !important"
    },
    link: {
      "&,&:hover,&:focus": {
        backgroundColor: "transparent",
        color: grayColor,
        boxShadow: "none"
      }
    },
    justIcon: {
      ...defaultFont,
      paddingLeft: "0px",
      paddingRight: "0px",
      height: "41px",
      minWidth: "41px",
      width: "41px",
      "& .fab,& .fas,& .far,& .fal,& svg,& .material-icons": {
        marginRight: "0px"
      },
      "&$lg": {
        height: "57px",
        minWidth: "57px",
        width: "57px",
        lineHeight: "56px",
        "& .fab,& .fas,& .far,& .fal,& .material-icons": {
          fontSize: "32px",
          lineHeight: "56px"
        },
        "& svg": {
          width: "32px",
          height: "32px"
        }
      },
      "&$sm": {
        height: "30px",
        minWidth: "30px",
        width: "30px",
        "& .fab,& .fas,& .far,& .fal,& .material-icons": {
          fontSize: "17px",
          lineHeight: "29px"
        },
        "& svg": {
          width: "25px",
          height: "25px",
          padding: "0.40625rem 0.40625rem"
        }
      }
    }
  };
}

export default buttonStyle;