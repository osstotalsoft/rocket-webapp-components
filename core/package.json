{
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "prebuild": "rimraf build",
    "build": "npm run build:cjs &&  npm run build:copy-files",
    "build:cjs": "babel src -d build --copy-files",
    "build:copy-files": "node ./scripts/copy-files.js",
    "bit-build": "bit -v && bit import && bit build"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.0",
    "@babel/core": "^7.7.7",
    "@babel/node": "^7.7.7",
    "@babel/plugin-proposal-decorators": "^7.8.3",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-react": "^7.8.0",
    "babel-eslint": "^10.0.3",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.7.0",
    "eslint-config-standard": "^14.1.0",
    "eslint-config-standard-react": "^9.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-node": "^11.0.0",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.17.0",
    "eslint-plugin-standard": "^4.0.1",
    "gh-pages": "^2.2.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^1.19.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "rimraf": "^3.0.0",
    "url-search-params-polyfill": "^7.0.1"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-flags": "^0.1.18",
    "react-router-dom": "^5.1.2"
  },
  "sideEffects": false,
  "dependencies": {},
  "bit": {
    "env": {
      "compiler": "totalsoft_oss.react-mui/compilers/react@1.0.32"
    },
    "componentsDefaultDirectory": "components/{name}",
    "packageManager": "npm"
  }
}
