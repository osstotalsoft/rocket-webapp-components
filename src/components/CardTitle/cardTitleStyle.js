import styles from "../../assets/jss/styles";
import defaultTheme from "../themes/defaultTheme";

const cardTitleStyle = theme => {
  const { defaultFont } = styles(theme);

  return {
    defaultFont,
    cardTitle: {
      ...(theme.header?.title || defaultTheme?.header?.title),
      display: "flex",
      flexGrow: 1,
      padding: "5px"
    },
    container: {
      display: "flex",
      width: "100%",
      flexWrap: "wrap",
      alignItems: "center",
      justifyContent: "flex-end"
    }
  };
};

export default cardTitleStyle;
