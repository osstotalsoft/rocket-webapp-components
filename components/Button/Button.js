import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { Button, makeStyles, Tooltip } from '@material-ui/core'
import buttonStyle from './buttonStyle'

const useStyles = makeStyles(buttonStyle)

function RegularButton({ tooltip, color, round, children, fullWidth, disabled, customClass, right, justIcon, size, wd, ...rest }) {
  const classes = useStyles()
  const btnClasses = cx({
    [classes[color]]: color,
    [classes.round]: round,
    [classes.fullWidth]: fullWidth,
    [classes.disabled]: disabled,
    [customClass]: customClass,
    [classes.right]: right,
    [classes.justIcon]: justIcon,
    [classes.wd]: wd,
    [classes[size]]: size
  })
  const classNames = `${classes.button} ${btnClasses}`

  return tooltip ? (
    <Tooltip title={tooltip}>
      <Button {...rest} className={classNames}>
        {children}
      </Button>
    </Tooltip>
  ) : (
    <Button {...rest} className={classNames}>
      {children}
    </Button>
  )
}

RegularButton.propTypes = {
  /**
   * The color of button.
   */
  color: PropTypes.oneOf([
    'primary',
    'info',
    'success',
    'warning',
    'danger',
    'rose',
    'theme',
    'themeNoBackground',
    'themeWithBackground',
    'defaultNoBackground',
    'primaryNoBackground',
    'infoNoBackground',
    'successNoBackground',
    'warningNoBackground',
    'dangerNoBackground',
    'roseNoBackground',
    'white',
    'simple',
    'transparent'
  ]),
  /**
   * If true, rounded corners are enabled.
   */
  round: PropTypes.bool,
  /**
   * 	The content of the component.
   */
  children: PropTypes.node,
  /**
   * 	If true, the button will take up the full width of its container.
   */
  fullWidth: PropTypes.bool,
  /**
   * If true, the button will be disabled.
   */
  disabled: PropTypes.bool,
  /**
   * A custom class provided.
   */
  customClass: PropTypes.string,
  /**
   * If true,  the button's min width will be set to 160px.
   */
  wd: PropTypes.bool,
  /**
   * If true, the button will be smaller.
   */
  justIcon: PropTypes.bool,
  /**
   * If true, the button will float to the right.
   */
  right: PropTypes.bool,
  /**
   * The size of the button.
   */
  size: PropTypes.oneOf(['sm', 'lg', 'xs']),
  /**
   * The tooltip of the button.
   */
  tooltip: PropTypes.string
}

export default RegularButton
