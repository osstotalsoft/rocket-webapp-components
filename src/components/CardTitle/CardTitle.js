import React from "react";
import PropTypes from "prop-types";
import cardTitleStyle from "./cardTitleStyle";
import { makeStyles } from "@material-ui/core/styles";
import { Box } from "@material-ui/core";

const useStyles = makeStyles(cardTitleStyle);

export default function CardTitle({ title, actions, ...rest }) {
  const classes = useStyles();

  return (
    <Box className={classes.container} >
      <Box className={classes.cardTitle}>{title}</Box>
      {actions.map(
        (action, index) => (
          <Box key={index} ml={1} {...rest}>
            {action}
          </Box>
        )
      )}
    </Box>
  );
}

CardTitle.propTypes = {
  title: PropTypes.string,
  actions: PropTypes.arrayOf(PropTypes.node).isRequired
};