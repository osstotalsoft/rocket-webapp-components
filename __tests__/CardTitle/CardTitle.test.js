import React from 'react';
import renderer from 'react-test-renderer';
import CardTitle from '../../components/CardTitle/CardTitle';
import Button from '../../components/Button/Button';
import { Box } from "@material-ui/core";

describe('CardTitle', () => {
    it('classes container and cardTitle are present', () => {
        const onClickMock = jest.fn();

        const wrapper = renderer.create(<CardTitle
            title="Add person"
            actions={[<Button key='button'
                onClick={onClickMock}
                disabled={true} />]}
        />);

        const testInstance = wrapper.root;

        expect(testInstance.findAllByType(Box)[0].props.className).toContain('container');
        expect(testInstance.findAllByType(Box)[1].props.className).toContain('cardTitle');
    });


    it('should be able to click the button', () => {
        const onClickMock = jest.fn();

        const wrapper = renderer.create(<CardTitle
            title="Add person"
            actions={[<Button key='button'
                onClick={onClickMock}
                disabled={true} />]}
        />);

        const testInstance = wrapper.root;

        testInstance.findAllByType(Box)[2].findByType(Button).props.onClick()
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });

    it('the button is expected to be disabled', () => {
        const onClickMock = jest.fn();

        const wrapper = renderer.create(<CardTitle
            title="Add person"
            actions={[<Button key='button'
                onClick={onClickMock}
                disabled={true} />]}
        />);

        const testInstance = wrapper.root;

        expect(testInstance.findAllByType(Box)[2].findByType(Button).props.disabled).toBe(true);
    });
});