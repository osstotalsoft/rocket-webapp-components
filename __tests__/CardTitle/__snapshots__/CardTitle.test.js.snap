// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 makeStyles-container-3"
>
  <div
    className="MuiBox-root MuiBox-root-5 makeStyles-cardTitle-2"
  >
    Add person
  </div>
  <div
    className="MuiBox-root MuiBox-root-6"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 makeStyles-disabled-27"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      />
    </button>
  </div>
</div>
`;
